### ISSUE TITLE
Can't call an API route from another route with auth iam enable in local

### ISSUE THREAD

**Issue by TheUncharted** [anomalyco/sst#5190]
In Local : I created an api gateway v2 and 2 routes one in typescript and one in python. The second route is protected with auth with IAM.
When the first route is invoked, it calls the second route with the sigv4 and the call fails because of an 403 error forbidden.

I use @aws-sdk/credential-provider-node, from the documentation : it will attempt to find credentials from the following sources (listed in order of precedence):

- Environment variables exposed via process.env
- SSO credentials from token cache
- Web identity token credentials
- Shared credentials and config ini files
- The EC2/ECS Instance Metadata Service

In my code , I use this block of code

```typescript
import { defaultProvider } from "@aws-sdk/credential-provider-node";

console.log("ENV", process.env);

const apiUrl = `${process.env.API_URL}/test2`;

const signer = new SignatureV4({
  credentials: defaultProvider(), //NOT WORKING
  // credentials: defaultProvider({profile: "default"}),
  sha256: Sha256,
  region: "eu-west-1",
  service: "execute-api",
});
```

By digging and displaying the signed request and also the environement variable, it seems that the calls use an access key id of type ASIA (Temporary Access Keys) that seems to be coming from the the environment variable of the lambda.
If I force the profile in the default provider, it works 

I'm not sure if it's a bug : shouldn't the access key be populated with your access key id in local ?
I tried also to specify the profile in the sst.config.ts but it seems the access key id in the environment variable is still an temporary one 

I reproduced on a small example with 2 typescript lambda :

[example](https://github.com/TheUncharted/test_lambda)

Currently, I invoke directly the lambda and its working so it's an urgent issue

**Comment by jayair** [comment]
So in `sst dev` it uses the credentials from the actual Lambda function instead of the local credentials. That's what you are seeing right?

**Comment by TheUncharted** [comment]
yes exactly, I don't know if it's intended but you can try with the basic sample I added

**Comment by jayair** [comment]
Yeah it is by design. We want to make sure it has the right permissions as it would on deploy.
